### PR TITLE
fix style in abstract-php*

### PR DESCRIPTION
**Pull request by yurikoles**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-php/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by yurikoles** [comment]
There are left some issues, I don't know how to fix:

```ruby
$ brew style --fix ./*
== abstract-php-extension.rb ==
W:115:  3: Avoid rescuing the Exception class. Perhaps you meant to rescue StandardError?
== abstract-php.rb ==
W:149:  5: Avoid rescuing the Exception class. Perhaps you meant to rescue StandardError?
W:301: 56: Use parentheses in the method call to avoid confusion about precedence.
W:306: 45: Use parentheses in the method call to avoid confusion about precedence.
```

**Comment by javian** [comment]
@yurikoles for the last two just adding the parentheses around the build.without call should be sufficient.

**Comment by yurikoles** [comment]
@javian fixed, thank you for suggestion.

File: Abstract/abstract-php-extension.rb [hunk 1]
```diff
@@ -20,29 +19,29 @@ class AbstractPhpExtension < Formula
...
 
-    if build.without? 'homebrew-php'
+    if build.without? "homebrew-php"
       installed_php_version = nil
       i = IO.popen("#{phpize} -v")
       out = i.readlines.join("")
       i.close
-      { 53 => 20090626, 54 => 20100412, 55 => 20121113, 56 => 20131106 }.each do |v, api|
-        installed_php_version = v.to_s if out.match(/#{api}/)
+      { 53 => 20_090_626, 54 => 20_100_412, 55 => 20_121_113, 56 => 20_131_106 }.each do |v, api|
```
Comment (javian) [inline Abstract/abstract-php-extension.rb]: If the idea is to make this more readable you should use the format that is intended. For 53 its 2009_06_26 indicating the 26th of June 2009. Writing it in the way you have now makes it more messy =)

File: Abstract/abstract-php.rb [hunk 2]
```diff
@@ -1,65 +1,70 @@
...
     head do
-      depends_on 'autoconf' => :build
-      depends_on 're2c' => :build
-      depends_on 'flex' => :build
-      depends_on 'bison@2.7' => :build
+      depends_on "autoconf" => :build
+      depends_on "re2c" => :build
+      depends_on "flex" => :build
+      depends_on "bison@2.7" => :build
+      depends_on "homebrew/versions/bison27" => :build
```
Comment (javian) [inline Abstract/abstract-php.rb]: This line should be removed - duplicate and incorrect.

File: Abstract/abstract-php.rb [hunk 3]
```diff
@@ -1,65 +1,70 @@
...
+      depends_on "autoconf" => :build
+      depends_on "re2c" => :build
+      depends_on "flex" => :build
+      depends_on "homebrew/versions/bison27" => :build
```
Comment (javian) [inline Abstract/abstract-php.rb]: This line should be removed - duplicate and incorrect.

File: Abstract/abstract-php.rb [hunk 4]
```diff
@@ -1,65 +1,70 @@
...
-      depends_on 're2c' => :build
-      depends_on 'flex' => :build
-      depends_on 'bison@2.7' => :build
+      depends_on "autoconf" => :build
+      depends_on "re2c" => :build
+      depends_on "flex" => :build
+      depends_on "bison@2.7" => :build
+      depends_on "homebrew/versions/bison27" => :build
+      depends_on "autoconf" => :build
+      depends_on "re2c" => :build
```
Comment (javian) [inline Abstract/abstract-php.rb]: Duplicate

File: Abstract/abstract-php.rb [hunk 5]
```diff
@@ -1,65 +1,70 @@
...
+      depends_on "flex" => :build
```
Comment (javian) [inline Abstract/abstract-php.rb]: duplicate

File: Abstract/abstract-php.rb [hunk 6]
```diff
@@ -1,65 +1,70 @@
...
-      depends_on 'autoconf' => :build
-      depends_on 're2c' => :build
-      depends_on 'flex' => :build
-      depends_on 'bison@2.7' => :build
+      depends_on "autoconf" => :build
+      depends_on "re2c" => :build
+      depends_on "flex" => :build
+      depends_on "bison@2.7" => :build
+      depends_on "homebrew/versions/bison27" => :build
+      depends_on "autoconf" => :build
```
Comment (javian) [inline Abstract/abstract-php.rb]: dupcliate

File: Abstract/abstract-php.rb [hunk 7]
```diff
@@ -209,173 +214,173 @@ def install_args
...
-      args << "--with-mssql=#{Formula['freetds'].opt_prefix}"
-      args << "--with-pdo-dblib=#{Formula['freetds'].opt_prefix}"
+    if build.with? "mssql"
+      args << "--with-mssql=#{Formula["freetds"].opt_prefix}"
+      args << "--with-pdo-dblib=#{Formula["freetds"].opt_prefix}"
     end
 
     # Do not build opcache by default; use a "phpxx-opcache" formula
-    args << "--disable-opcache" if php_version.start_with?('5.5', '5.6', '7.0', '7.1')
+    args << "--disable-opcache" if php_version.start_with?("5.5", "5.6", "7.0", "7.1")
```
Comment (javian) [inline Abstract/abstract-php.rb]: You can change the last two arguments into a single "7."

**Comment by javian** [comment]
Can I also ask you to bump the revision of php71 just so that we know php will compile with these changes ?

**Comment by yurikoles** [comment]
@javian done with success.

**Comment by javian** [comment]
Check https://bot.brew.sh/job/Homebrew%20PHP%20Pull%20Requests/2910/version=sierra/console - something has gone wrong in the file making everything break.

**Comment by javian** [comment]
You seem to have removed an end statement here https://github.com/Homebrew/homebrew-php/pull/4063/files#diff-4867fde2318630c52b20c12ef35df108L30

**Comment by yurikoles** [comment]
@javian nice catch, thank you!

**Comment by javian** [comment]
As far as the exception warning then maybe this will shed some light on it http://blog.honeybadger.io/ruby-exception-vs-standarderror-whats-the-difference/

**Comment by javian** [comment]
I'd say lets merge this now and you can open a new PR since it doesn't affect anything and it does break anything.

**Comment by javian** [comment]
Thanks so far @yurikoles !

**Comment by yurikoles** [comment]
@javian you welcome

**Pull request closed without merging**